x/evm/keeper/grpc_query.go

```diff
@@ -341,8 +352,28 @@ func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*type
 			msg.IsFake(),
 		)
 
+		tmpCtx := ctx
+		if fromType == FromTypeRPC {
+			ctx = ctx.WithGasMeter(sdk.NewGasMeter(msg.Gas())).WithKVGasConfig(storetypes.GasConfig{}).
+				WithTransientKVGasConfig(storetypes.GasConfig{})
+			tmpCtx, _ := ctx.CacheContext()
+
+			acct := k.GetAccount(tmpCtx, msg.From())
+
+			from := msg.From()
+			if acct == nil {
+				acc := k.accountKeeper.NewAccountWithAddress(tmpCtx, from[:])
+				k.accountKeeper.SetAccount(tmpCtx, acc)
+				acct = statedb.NewEmptyAccount()
+			}
+			acct.Nonce = nonce + 1
+			err = k.SetAccount(tmpCtx, from, *acct)
+			if err != nil {
+				return true, nil, err
+			}
```

Comment:
Why is this needed?

Reply:
in really  processing msg,  before apply message, nonce will incr in [here](https://github.com/evmos/evmos/blob/469c5cc77be774441b9fa8ba013a2d73140a0e13/app/ante/evm/eth.go#L386).
if remove this logic, unittest will failed.

Reply:
with these changes, I see the staking test is failing when estimating the gas with the error

```
rpc error: code = Unknown desc = no delegation distribution info
```

Without these changes, the test passes

Reply:
I am trying to reproduce the err you mentioned. please tell me more. like what command you run to cause the error.

Reply:
just run the `test_precompiles.py` test in the nix-shell

Reply:
setting the gasMeter after increasing the sequence fixes the issue:
```
		if fromType == FromTypeRPC {
			tmpCtx, _ = ctx.CacheContext()

			acct := k.GetAccount(tmpCtx, msg.From())

			from := msg.From()
			if acct == nil {
				acc := k.accountKeeper.NewAccountWithAddress(tmpCtx, from[:])
				k.accountKeeper.SetAccount(tmpCtx, acc)
				acct = statedb.NewEmptyAccount()
			}
			acct.Nonce = nonce + 1
			err = k.SetAccount(tmpCtx, from, *acct)
			if err != nil {
				return true, nil, err
			}
			tmpCtx = tmpCtx.WithGasMeter(evmostypes.NewInfiniteGasMeterWithLimit(msg.Gas())).WithKVGasConfig(storetypes.GasConfig{}).
			WithTransientKVGasConfig(storetypes.GasConfig{})
		}
```

Reply:
Please add comments to the code to explain why we're increasing the nonce/sequence

Reply:
And why we need to reset the gas meter there

Reply:
done, and thanks.